py3status/modules/usbguard.py

```diff
@@ -200,182 +82,94 @@ def post_config_hook(self):
 
         format_device_separator = self.py3.safe_format(self.format_device_separator)
-        format_device = self.py3.composite_join(format_device_separator, new_device)
-
-        return format_device, format_device_list
+        format_device = self.py3.composite_join(format_device_separator, device_info)
 
-    def _notify_user(self, device):
-        format_notification_message = self.py3.safe_format(
-            self.format_notification_message, device
-        )
-        format_notification_title = self.py3.safe_format(
-            self.format_notification_title, device
-        )
-        self.py3.notify_user(
-            msg=format_notification_message,
-            title=format_notification_title,
-            icon="/usr/share/icons/hicolor/scalable/apps/usbguard-icon.svg",
-        )
+        return format_device
 
     def usbguard(self):
-        filter_name = self.filters[self.filter_index]
-        device_data = self._get_devices(filter_name)
-        format_device, format_device_list = self._manipulate_devices(
-            device_data, filter_name
-        )
-
-        if self.device_index > self.device_length:
-            self.device_index = self.device_length
-
-        format_button_device = self.py3.safe_format(
-            self.format_button_device,
-            {"device": self.device_index, "devices": self.device_length},
-        )
-        self.py3.composite_update(format_button_device, {"index": "button_device"})
-
-        format_button_filter = self.py3.safe_format(
-            self.format_button_filter, {"filter": filter_name}
-        )
-        self.py3.composite_update(format_button_filter, {"index": "button_filter"})
-        format_button_permanent = self.py3.safe_format(self.format_button_permanent)
-        self.py3.composite_update(
-            format_button_permanent, {"index": "button_permanent"}
-        )
+        self.devices = self._get_devices()
 
         usbguard_data = {
-            "device": len(device_data),
-            "format_button_filter": format_button_filter,
-            "format_button_permanent": format_button_permanent,
-            "format_device": format_device,
-            "format_button_device": format_button_device,
-            "format_device_list": format_device_list,
+            "device": len(self.devices),
+            "format_device": self._format_device(self.devices),
         }
 
-        response = {
-            "cached_until": self.py3.time_in(10),
+        return {
+            "cached_until": self.py3.CACHE_FOREVER,
             "full_text": self.py3.safe_format(self.format, usbguard_data),
+            "urgent": True,
         }
 
-        if self.allow_urgent:
-            response["urgent"] = True
-
-        self.device_data = device_data
-        return response
-
     def on_click(self, event):
-        index = event["index"]
-        button = event["button"]
-        if isinstance(index, int):
+        if not isinstance(event["index"], str):
```

Comment:
IIRC `str` not supported on Python2.

Reply:
I was about to loudly "booooo", but this time we are in luck:

```
❯ python2                         
Python 2.7.15 (default, Jan 10 2019, 23:20:52) 
[GCC 8.2.1 20181127] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> s = 'a string'
>>> if isinstance(s, str):
...   print('supported!!!')
... 
supported!!!
>>> 
```

Reply:
```diff
diff --git a/py3status/modules/static_string.py b/py3status/modules/static_string.py
index 1cfbfed..6f06381 100644
--- a/py3status/modules/static_string.py
+++ b/py3status/modules/static_string.py
@@ -17,9 +17,10 @@ class Py3status:
     """
 
     # available configuration parameters
-    format = "Hello, world!"
+    format = u"_▁▂▃▄▅▆▇█"
 
     def static_string(self):
+        self.py3.log(isinstance(self.format, str))
         return {
             "cached_until": self.py3.CACHE_FOREVER,
             "full_text": self.py3.safe_format(self.format),
```
```bash
$ python3 static_string.py
Module `test_module`: True
{'full_text': '_▁▂▃▄▅▆▇█', 'cached_until': -1}
^C%

$ python2 static_string.py
Module `test_module`: False
{'full_text': u'_\u2581\u2582\u2583\u2584\u2585\u2586\u2587\u2588', 'cached_until': -1}
^C%
```

Reply:
OMG, boooooooo python2! I hate it, kills so many cool ideas :(

Will revert to a less error-prone `int` comparison in a second.